`jmwalletd`: `coinjoin_in_process` remains `true` after taker operation fails

When a taker operations fails, in this example with `txn-mempool-conflict`, the `coinjoin_in_process` of `/session` endpoint remains `true` forever.

Expected: When a taker operation fails, it should indicate with `coinjoin_in_process := false` in `/session`
Actual: When a taker operation fails, `/session` reports `coinjoin_in_process := true` forever

Steps to reproduce:
1. Start a taker operation `POST /taker/coinjoin` (e.g. sending `499999999` to `bcrt1qact...64qj`)
2. Send direct payment `/taker/direct-send` after "Commitment sourced OK" (e.g. sending `500000001` to `bcrt1q7m2...dmtz`)
3. `coinjoin_in_process` of `/session` stays `true` forever

Logs (from network `regtest`):
```
2022-03-27 02:47:00,720 [INFO]  INFO:Commitment sourced OK
[...]
2022-03-27 02:47:01,415 [INFO]  Sends: 5.00000001 BTC (500000001 sat) to destination: bcrt1q7m2d09tscne4sqm73q4lm89mtfgx5ah2tvdmtz
2022-03-27 02:47:01,416 [DEBUG]  rpc: sendrawtransaction ['...']
2022-03-27 02:47:01,418 [INFO]  Transaction sent: 635e7b585cfaf0afc9ea06d53f3667b477569105d46454c8228d7abead33815a
2022-03-27 02:47:02,929 [INFO]  Removed utxos=
27388c6f6949b988c864289457aa4a4b80549819d35c403a0b9aa0d3ae124766:0 - path: m/84'/1'/0'/0/0, address: bcrt1qurjlxatvmlwygyjkup7nsm0xp4np22zf5tlpr5, value: 5000000000
2022-03-27 02:47:02,932 [INFO]  Added utxos=
635e7b585cfaf0afc9ea06d53f3667b477569105d46454c8228d7abead33815a:0 - path: m/84'/1'/0'/1/1, address: bcrt1qfc4ns9lnkmhulwlfvn6m8evp8txsy8dalawgqe, value: 4499998579
635e7b585cfaf0afc9ea06d53f3667b477569105d46454c8228d7abead33815a:1 - path: m/84'/1'/4'/0/0, address: bcrt1q7m2d09tscne4sqm73q4lm89mtfgx5ah2tvdmtz, value: 500000001
[...]
2022-03-27 02:47:04,753 [INFO]  obtained tx
2022-03-27 02:47:04,753 [INFO]  INFO:Built tx, sending to counterparties.
2022-03-27 02:47:06,104 [DEBUG]  found good sig at index=1
2022-03-27 02:47:06,105 [DEBUG]  nick = J52cn6FBqvmuShKx sent all sigs, removing from nonrespondant list
2022-03-27 02:47:06,105 [INFO]  all makers have sent their signatures
2022-03-27 02:47:06,105 [INFO]  INFO:Transaction is valid, signing..
[...]
2022-03-27 02:47:06,110 [INFO]  txid = 6e89f25b3a88a6c8f043470bc4d65d546904333dd6abb9ccb8eed82a7b96df24
2022-03-27 02:47:06,111 [DEBUG]  rpc: sendrawtransaction ['...']
2022-03-27 02:47:06,112 [WARNING]  error pushing = -26 txn-mempool-conflict
[...]
2022-03-27 02:50:06,112 [INFO]  Timed out: transaction with txid: 6e89f25b3a88a6c8f043470bc4d65d546904333dd6abb9ccb8eed82a7b96df24 not broadcast.
2022-03-27 02:50:06,115 [DEBUG]  rpc: sendrawtransaction ['...']
2022-03-27 02:50:06,118 [WARNING]  error pushing = -26 txn-mempool-conflict
2022-03-27 02:50:06,119 [ERROR]  Failed to broadcast transaction: 
```

After that, `/session` will always respond with:
```json
{"session": true, "maker_running": false, "coinjoin_in_process": true, "wallet_name": "funded.jmdat"}
```

Workaround: if the wallet is locked and unlocked, the state is reported correctly.
Should the `coinjoin_in_progress` flag revert to `false` after the operation fails?

What are your thoughts on this?

--- a/jmclient/jmclient/wallet_rpc.py
+++ b/jmclient/jmclient/wallet_rpc.py
@@ -431,18 +431,11 @@ class JMWalletDaemon(Service):
         # It is also different in that the event loop must not shut down
         # when processing finishes.
 
-        # This assertion is making sure that the callback is not a non-final
-        # schedule entry (see the comment above as to why). We accept
-        # either False or 'unconfirmed' since they can arrive in either order.
-        assert fromtx is not True
-
         # reset our state on completion, we are no longer coinjoining:
         self.taker = None
 
         if not res:
             jlog.info("Coinjoin did not complete successfully.")
-        #Should usually be unreachable, unless conf received out of order;
-        #because we should stop on 'unconfirmed' for last (see above)
         else:
             jlog.info("Coinjoin completed correctly")
 

